docs/jwst/ramp_fitting/main.rst

```diff
@@ -16,34 +16,36 @@ detailed below.
-using the 4-D GROUPDQ array of the input data set, under the assumption that
-the jump step will have already flagged CR's. Ramp intervals are also terminated
-where saturation flags are found.  Ramp intervals that are noiseless, or have
-no signal, or contain only 2 reads will initially have fits with variance = 0,
-preventing their slopes from contributing to the weighted slopes.  In these
-cases, the variance will be recalculated as the poisson noise of the ramp added
-in quadrature to the pixel-specific read noise, ensuring that all variance
-values are positive.  If the input dataset has only a single group in each
-integration, the count rate for all unsaturated pixels in that integration will
-be calculated to be the value of the science data in that group divided by the
-exposure time.  If the input dataset has only two groups per integration, the
-count rate for all unsaturated pixels in each integration will be calculated
-from the 2 valid values of the science data.  If any input dataset contains
-ramps saturated in their second read, the count rates for those pixels in that
-integration will be calculated to be the value of the science data in that group
-divided by the exposure time. After computing the slopes for all intervals for
-a given pixel, the final slope is determined as a weighted average from all
-intervals and is written to a file as the primary output product.  In this
-output product, the 4-D GROUPDQ from all integrations is compressed into 2-D,
-which is then merged (using a bitwise OR) with the input 2-D PIXELDQ to create
-the output DQ array.  The 3-D VAR_POISSON and VAR_RNOISE arrays from all 
-integrations are averaged into corresponding 2-D output arrays.  If the ramp 
-fitting step is run by itself, the output file name will have the suffix 
-'_RampFit' or the suffix '_RampFitStep'; if the ramp fitting step is run as part 
-of the calwebb_detector1 pipeline, the final output file name will have the 
-suffix '_rate'.  In either case, the user can override this name by specifying 
-an output file name.
+cosmic-ray-free ramp intervals for each pixel. The fitting algorithm does an 
+'optimal' linear fit, which is the weighting used by Fixsen et 
+al, PASP,112, 1350. ('unweighted' in which pixels are equally weighted, is no 
+longer a weighting option.)  CR-free intervals are derived using the 4-D
+GROUPDQ array of the input data set, under the assumption that the jump step
+will have already flagged CR's. Ramp intervals are also terminated where
+saturation flags are found. Pixels are processed simultaneously in blocks 
+using the array-based functionality of numpy.  The size of the block depends
+on the image size and the number of groups.
+
+
+If the input dataset has only a single group in each integration, the count rate
+for all unsaturated pixels in that integration will be calculated to be the
+value of the science data in that group divided by the exposure time.  If the
```

Comment:
The slope should be the value in that group divided by the integration time (or the group time), not the exposure time.